crates/apollo-mcp-registry/src/platform_api/operation_collections/collection_poller.rs

```diff
@@ -60,12 +60,11 @@ async fn handle_poll_result(
     poll: Vec<(String, String)>,
     platform_api_config: &PlatformApiConfig,
 ) -> Result<Option<Vec<OperationData>>, CollectionError> {
-    let mut keep_ids = poll.iter().map(|(id, _)| id);
-    for id in previous_updated_at.clone().keys() {
-        if keep_ids.all(|keep_id| keep_id != id) {
-            previous_updated_at.remove(id);
-        }
-    }
+    let removed_ids = previous_updated_at.clone();
+    let removed_ids = removed_ids
+        .keys()
+        .filter(|id| poll.iter().all(|(keep_id, _)| keep_id != *id))
+        .collect::<Vec<_>>();
```

Comment:
It seems like an unnecessary clone.

```suggestion
    let removed_ids: Vec<&String> = previous_updated_at
        .keys()
        .filter(|id| poll.iter().all(|(keep_id, _)| keep_id != *id))
        .collect();
```

Reply:
If I do that I get errors later on telling me previous_updated_at is borrowed :(